### DISCUSSION TITLE
Preserving repeated frames when using "-layers compare-any"

### DISCUSSION THREAD

**Discussion by JasonMDavey** [ImageMagick/ImageMagick#2616] [Help]
I'm creating GIFs from sequences of PNGs, like this:
`convert -delay 5 -loop 0 -layers compare-any "image*.png" output.gif`

(I'm using compare-any because the context in which these are being displayed does not properly render GIFs created with "optimize", otherwise I'd just use that).

The issue I'm having is that when the input images contain sequences of identical frames, those are all being collapsed into one frame in the output. I would expect the frame duration in the output to be extended to account for these duplicates? But instead they are just run together, so the timing is not preserved properly.

I've worked around this by manually introducing minute differences (tweaking a single pixel) to ensure that there are no exact duplicates, but this results in a significantly larger output filesize.

Is there anything else I can do here? Thanks!

**Comment by snibgo** [comment]
Why doesn't "-layers optimize" work for you? For other possibilities, see https://imagemagick.org/script/command-line-options.php#layers . Perhaps "-layers remove-dups" does what you want.

**Reply by JasonMDavey** [reply]
"-layers optimize" doesn't work because, presumably due to a bug on their end, Discord (popular messaging app where I need to display these images) does not render GIFs created with this option correctly (there is visual corruption).

"remove-dups" produces the expected output visually, but with significantly higher filesize (because it's not doing any crop-based optimisation).

Basically I want the optimisation which "compare-any" provides (as this results in small filesizes), along with the time delay merging which "remove-dups" provides (as this preserves playback timing properly). I'm surprised that "compare-any" doesn't do this by default - is it expected behaviour that "compare-any" discards duplicate frames without extending the duration to account for the removed frames?

Thanks for the response! I think I've actually found a workaround with the following command (remove-dups sorts out the timing, and compare-any at least preserves these merged time delays, so we see the correct result).

`convert -delay 5 -loop 0 -layers remove-dups -layers compare-any "image*.png" output.gif`
